pyfr/integrators/dual/pseudo/pseudocontrollers.py

```diff
@@ -140,8 +145,27 @@ def __init__(self, *args, **kwargs):
                     )
                 )
 
+        self.pseudo_step_multiplied(1.0)
+
         self.backend.commit()
 
+    def adjust_pseudo_step(self, dt):
+        old_dtau = self.dtau
+        super().adjust_pseudo_step(dt)
```

Comment:
This is maybe not the most straightforward way of implementing this. It adds spaghetti that will make it hard in the future to refactor. Consider inlining the `super().adjust_pseudo_step()`?